### PR TITLE
[None][refactor] add Vanilla sparse attention primitives

### PR DESCRIPTION
**Pull request by yihwang-nv**
Add shared Vanilla primitives for selected sparse GQA and MLA.

- Added shared Vanilla primitives for sparse one-token GQA and selected MLA attention.
- Added input and dimension validation.
- Filters padded indices and deduplicates sparse KV blocks.
- Supports per-KV-head selection, softmax scoring, value aggregation, and optional attention sinks.
- No public API or configuration changes.

## Dev Engineer Review

- Added sparse one-token GQA and selected MLA attention primitives.
- Added input and dimension validation.
- Added padded-index filtering, KV-block deduplication, per-KV-head selection, float32 softmax scoring, value aggregation, and optional attention-sink normalization.
- No public API, configuration, or test-list changes identified.
- Numerical stability, validation behavior, performance, and coding-guideline consistency require review.
- CI pipelines `#55553`, `#55593`, `#56180`, and `#56231` failed. Run `#68724` was aborted. Run `#68735` failed.

## QA Engineer Review

- Added `test_sparse_gqa_deduplicates_blocks`.
- Added `test_selected_mla_attention_sink`.
- Neither test is covered in `tests/integration/test_lists/` test-db or qa files.
- **Verdict: insufficient.**

### CODE REVIEW

File: tensorrt_llm/_torch/attention_backend/vanilla.py [hunk 1]
```diff
@@ -348,6 +348,55 @@ def _single_request_attn_forward(self,
...
+            selected_keys = key_states.index_select(
+                0, token_indices)[:, kv_head].to(torch.float32)
+            selected_values = value_states.index_select(
+                0, token_indices)[:, kv_head].to(torch.float32)
+            head_start = kv_head * heads_per_kv
+            head_end = head_start + heads_per_kv
+            scores = torch.matmul(q[head_start:head_end].to(torch.float32),
+                                  selected_keys.T) * qk_scale
+            probabilities = F.softmax(scores, dim=-1, dtype=torch.float32)
+            output[head_start:head_end] = torch.matmul(probabilities,
```
Comment (yuxianq) [inline tensorrt_llm/_torch/attention_backend/vanilla.py]: Can we use the existed `torch.nn.functional.scaled_dot_product_attention` instead of combination of matmul + softmax + matmul here?

**Review by yuxianq — APPROVED** [review]

File: tensorrt_llm/_torch/attention_backend/vanilla.py (hunk 1 above)
Comment (yihwang-nv) [reply tensorrt_llm/_torch/attention_backend/vanilla.py]: Fixed
